src/web/browserCommon.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Returns the browser Id and version from the user agent string
+ * @param userAgent
+ */
+export function browserIdFromAgent(userAgent: string): BrowserResult {
+  if (!userAgent) {
+    return { id: BrowserIdValues.UnknownAgent, version: "" };
+  }
+
+  let matches = userAgent.match(/msie\/([0-9\.]+)/i);
```

Comment:
The regular expressions inside this function create a new Regex instance for each one, every time the function is called.

As an optimizaction, shouldn't we define de regular expressions as constants outside the function?

Reply:
Fixed at [bae20d5](https://github.com/genexuslabs/web-standard-functions/pull/77/commits/bae20d534cfbf14e2628c474430922d670bd0245)